File: python/Qwords/models/Word.py
import logging
logger = logging.getLogger(__name__)

class Word(object):
    def __init__(self, word: str) -> None:
        self.value = word.strip().lower()

    def get_word(self) -> str:
        return self.value
    
    def contains(self, c: str) -> bool:
        return c in self.word
    
    def is_correct(self, guess:list) -> bool:
        return self.word == guess
    
    def get_info(self, guess: str) -> str:
        if not guess:
            return ""
        
        info = []
        guess = guess.lower()
        word = self.value
        
        logger.debug(f"Word: {word}")
        logger.debug(f"Guess: {guess}")
        
        for i, c in enumerate(guess):
            if i >= len(word):
                info.append('X')  # Invalid position
            elif c == word[i]:
                info.append('+')  # Correct position
            elif c in word:
                info.append('?')  # Wrong position but in word
            else:
                info.append('✗')  # Not in word
        
        result = ''.join(info)
        logger.debug(f"Info: {result}")            
        return result

File: csharp/QWORDSCSHARP/Models/Word.cs
using System.Text;

public class Word
{
    private string word;
    private char[] characters;

    public Word(string word)
    {
        this.word = word;
        this.characters = word.ToCharArray();
    }

    public string GetWord()
    {
        return word;
    }

    public bool Contains(char c)
    {
        return characters.Contains(c);
    }

    public bool IsCorrect(char[] guess)
    {
        return new string(guess).Equals(word, StringComparison.OrdinalIgnoreCase);
    }

    public string GetInfo(string guess)
    {
        var guessArray = guess.ToCharArray();
        var result = new StringBuilder();

        for (int i = 0; i < guessArray.Length; i++)
        {
            char currentGuess = guessArray[i];
            if (Contains(currentGuess))
            {
                if (characters[i] == currentGuess)
                {
                    result.Append('+');
                }
                else
                {
                    result.Append('?');
                }
            }
            else
            {
                result.Append('X');
            }
        }

        return result.ToString();
    }
}
